compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingInterfaces.td

```diff
@@ -90,4 +94,38 @@ def IREEEncoding_EncodingLayoutAttrInterface :
   ];
 }
 
+//===----------------------------------------------------------------------===//
+// Type Interfaces
+//===----------------------------------------------------------------------===//
+
+def IREEEncoding_EncodingTypeInterface :
+  TypeInterface<"EncodingTypeInterface"> {
+  let cppNamespace = "::mlir::iree_compiler::IREE::Encoding";
+
+  let description = [{
+    Interface used to access/update tensor types with encodings.
+  }];
+
+
+  let methods = [
+    InterfaceMethod<
+      [{
+        Returns the tensor type with the encoding.
+      }],
+      /*retTy=*/"::mlir::Type",
+      /*methodName=*/"getEncodingType",
+      /*args=*/(ins)
+    >,
+    InterfaceMethod<
+      [{
+        Returns the same type but with the updated encoding.
+      }],
+      /*retTy=*/"::mlir::Type",
```

Comment:
Also here

Reply:
Mostly because of current needs and plugin needs. E.g., we have `Flow::DispatchTensorType` in IREE and it is not a `RankedTensorType`. The non-builtin type can return whatever types. In the current use case, the `Flow::DispatchTensorType` returns the `Flow::DispatchTensorType` with the updated encodings.